internal/bytes/string.go

```diff
@@ -0,0 +1,26 @@
+package common
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+// UnsafeStrToBytes uses unsafe to convert string into byte array. Returned bytes
+// must not be altered after this function is called as it will cause a segmentation fault.
+func UnsafeStrToBytes(s string) []byte {
+	var buf []byte
```

Comment:
What do you think about the following to avoid copying the slice internals by directly casting the string:
```
func unsafeGetBytes(s string) []byte {
    return (*[0x7fff0000]byte)(unsafe.Pointer(
        (*reflect.StringHeader)(unsafe.Pointer(&s)).Data),
    )[:len(s):len(s)]
}
```
I'm not sure if there are any constraints preventing that but this could be a further optimization

Reply:
What is `[0x7fff0000]byte`? Looks like a big static array. Where did you get that solution from?

Reply:
It is the maximum [process address space](https://stackoverflow.com/questions/27675550/what-is-the-address-space-in-golang) value

I grabbed this suggestion from here:
https://stackoverflow.com/questions/59209493/how-to-use-unsafe-get-a-byte-slice-from-a-string-without-memory-copy

Original source: https://groups.google.com/g/golang-nuts/c/Zsfk-VMd_fU/m/O1ru4fO-BgAJ